### PR TITLE
Update peano linker script generation

### PR DESCRIPTION
**Pull request by stephenneuendorffer**
It was possible for some input sections to not get properly copied to our executable.  Fix the linker script and enable more error checking during linking to ensure this doesn't happen.

### CODE REVIEW

File: python/compiler/aiecc/main.py [hunk 1]
```diff
@@ -465,7 +465,10 @@ async def process_core(
                 install_path, "aie_runtime_lib", aie_target.upper()
             )
 
-            clang_link_args = ["-Wl,--gc-sections"]
+            # --gc-sections to eliminate unneeded code.
+            # --orphan-handling=error to ensure that the linker script is as expected.
+            # If there are orphaned input sections, then they'd likely end up outside of the normal program memory.
+            clang_link_args = ["-Wl,--gc-sections", "--orphan-handling=error"]
```
Comment (jgmelber) [inline python/compiler/aiecc/main.py:471]: `"--orphan-handling=error"` threw an error locally for me: `error: unknown argument: '--orphan-handling=error'`

**Comment by jgmelber** [comment]
I was using this PR for Peano: https://github.com/Xilinx/llvm-aie/pull/142

File: python/compiler/aiecc/main.py (hunk 1 above)
Comment (stephenneuendorffer) [reply python/compiler/aiecc/main.py]: Sorry, typo on my part

**Review by jgmelber — APPROVED** [review]
Tested this works for compiling a C++ kernel without AIE API using Peano PR: https://github.com/Xilinx/llvm-aie/pull/142

**Pull request merged**